cmd/gcp-controller-manager/dpwi/nodesyncer/nodesyncer.go

```diff
@@ -0,0 +1,249 @@
+package nodesyncer
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+	"sort"
+	"strings"
+	"sync"
+	"time"
+
+	core "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	coreinformers "k8s.io/client-go/informers/core/v1"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/cache"
+	clientcmdapi "k8s.io/client-go/tools/clientcmd/api"
+	"k8s.io/client-go/util/workqueue"
+	"k8s.io/cloud-provider-gcp/cmd/gcp-controller-manager/dpwi/hms"
+	"k8s.io/cloud-provider-gcp/cmd/gcp-controller-manager/dpwi/serviceaccounts"
+	"k8s.io/klog/v2"
+)
+
+const (
+	indexByNode = "index-by-node"
+)
+
+// nodeSyncer implements a custom control loop responsible for synchronizing GCE with the list of
+// authorized and in-use GCP Service Accounts (GSA) on each Node.  GSA authorization is handled by
+// the service-account-verifier control loop which shares the result over the verifiedSAs map.  Use
+// of individual authorized GSA on each node is tracked by this control loop based on Pod events.
+type nodeSyncer struct {
+	indexer   cache.Indexer
```

Comment:
We should call this podIndexer. The type isn't clear because it's in a struct called nodeSyncer.

Reply:
Changed, thank you.